### PR TITLE
refactor(web): split convert() and api_v1_convert() into validate/enqueue/respond helpers

### PR DESCRIPTION
**Pull request by aerocristobal**
## Summary
- `convert()` (~89 lines) and `api_v1_convert()` (~141 lines) in `web/routes/conversion.py` mixed request validation, job creation, and Celery dispatch in single monolithic functions.
- Splits each into named helpers with the same responsibility split for both routes:
  - `_validate_convert_request` / `_validate_convert_file` / `_enqueue_convert_job` / `_respond_convert_success` (Web UI `/convert`)
  - `_validate_v1_convert_params` / `_resolve_v1_convert_format` / `_enqueue_v1_convert_job` / `_respond_v1_convert_success` (`/api/v1/convert`)
- Pure behavior extraction — every validation branch, error message, status code, and side-effect ordering (including the pre-existing quirk where `api_v1_convert` creates the job's storage dirs before running content-type validation) is preserved exactly as it was.

## Test plan
- [x] Full existing `test_web.py` + `test_api_v1.py` suite (107 tests covering both routes' happy paths, every validation error, pandoc_options edge cases, and auth) passes unmodified against the split code
- [x] 33 new isolated unit tests (`test_conversion_route_helpers.py`) call each helper directly with crafted inputs — no HTTP request needed
- [x] 140 passed total, no regressions

Story 6.4a (docs/BACKLOG.md, Story 6.4 decomposed)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: web/routes/conversion.py [hunk 1]
```diff
@@ -572,6 +617,47 @@
...
+    use_llm = request.form.get('use_llm', 'false').lower() == 'true'
+    raw_pandoc = request.form.get('pandoc_options')
+
+    error, pandoc_options = _validate_v1_convert_params(to_format, engine, raw_pandoc)
+    if error:
+        return error
+
+    error, from_format, internal_from_format = _resolve_v1_convert_format(file.filename, from_format, engine)
+    if error:
+        return error
```
[the remark this answers is not in the document]
Comment (aerocristobal) [reply web/routes/conversion.py:664]: Resolved. All user-provided values (, , , ) are now escaped via  before interpolation into error messages. CodeQL scan passed on the updated branch.
